code/__HELPERS/icons.dm

```diff
@@ -773,13 +773,8 @@ world
 
 	//Try to remove/optimize this section ASAP, CPU hog.
 	//Determines if there's directionals.
-	if(render_icon && curdir != SOUTH)
-		if (
-			!length(icon_states(icon(curicon, curstate, NORTH))) \
-			&& !length(icon_states(icon(curicon, curstate, EAST))) \
-			&& !length(icon_states(icon(curicon, curstate, WEST))) \
-		)
-			base_icon_dir = SOUTH
+	if(render_icon && curdir != SOUTH && !length(icon_states(icon(curicon, curstate, curdir))))
```

Comment:
what happens if curdir is a diagonal

Reply:
if the diagonal has a icon state that gets used, if it doesn't it defaults to south, this is the same functionality but only checks icon states once instead of three times

Reply:
yeah but stuff like northeast defaults to east if there is no northeast icon, not south

Reply:
thanks I hate it

Reply:
```suggestion
if(render_icon && curdir != SOUTH)
		if (
			!length(icon_states(icon(curicon, curstate, NORTH))) \
			&& !length(icon_states(icon(curicon, curstate, EAST))) \
			&& !length(icon_states(icon(curicon, curstate, WEST))) \
		)
			base_icon_dir = SOUTH
```